python/ray/data/datasource/_default_metadata_providers.py

```diff
@@ -13,7 +13,7 @@ def get_generic_metadata_provider(file_extensions: Optional[List[str]]):
     return DefaultFileMetadataProvider()
 
 
-def get_parquet_metadata_provider():
```

Comment:
or pass in the ParquetDatasource as the parameter?

Reply:
> actually, would it be better to add all read_parquet parameters?

What would the motivation for this? To make the signature less arbitrary?

> or pass in the ParquetDatasource as the parameter?

`ParquetDatasource` doesn't have the value of `override_num_blocks`, so I'm not sure if that'd work.

Reply:
> What would the motivation for this? To make the signature less arbitrary?
just in case we'll need more info for determining metadata provider in the future

Reply:
Gotcha. Given that we might not even use the `_default_metadata_provider` approach going forward, let's revisit later